### DISCUSSION TITLE
using @use rule with Bootstrap

### DISCUSSION THREAD

**Discussion by mysticCoder100** [twbs/bootstrap#41088] [Q&A] [answered]
I've noticed that Bootstrap hasn't implemented the `@use` rule yet. Are there any workarounds or alternative approaches to using Bootstrap components in a modular way with Sass?

**Comment by mysticCoder100** [comment]
Is there no way to use the @use rule or the bootstrap team would implement using the @forward and @use rules in later version.

**Comment by mdo** [comment]
Long story short, it's very difficult to migrate as-is.

**Reply by mrholek** [reply]
I believe migrating it without some changes in the source code is not possible. I began the migration, but the first obstacle I encountered was that `_variables.scss` required certain functions from `_functions.scss`. Additionally, some functions from `_functions.scss`, like `color-contrast`, require specific variables, so some functions need to be rewritten. Another issue is that mixins with an underscore, such as `_assert-ascending`, do not work.

I’m not sure if you will permit the following changes in v5, but if you do, I can help you because I need to implement this migration in my Bootstrap-based project https://github.com/coreui/coreui due to the numerous support requests I've received about this Sass warning.

**Reply by julien-deramond** [reply, marked as answer]
I'm currently tackling it in https://github.com/twbs/bootstrap/pull/41112 for a possible integration in the last Bootstrap v5 release (maybe 5.5.0). This is one of the next steps. When it'll be more stable, you could help with the testing phase.
